### ISSUE TITLE
Add role prop to Message component

### ISSUE THREAD

**Issue by WillSmithTE** [WillSmithTE/carbon#7]
### Description

The Message component currently does not accept a role. Different roles can be applied to the Message to help with screen reader behaviour.

### Suggested solution

Accept a role prop to allow products to add `role="alert"` or `role="status"` for example

### Confidentiality

- [x] I confirm there is no confidential or commercially sensitive information included.

**Comment by WillSmithTE** [comment]
**nuria1110** commented on 2026-02-11T15:09:04Z:

---

Note for triage:
These roles add an implicit aria-live to the element. The reason for this request was so that screen readers announce non-intrusive messages like "Success" without having to pull focus to the Message and disrupting user navigation. 

We are not be able to pass the role directly onto the component wrapper since it is conditionally rendered and screen readers will not announce elements with aria-live that are dynamically added to the DOM. 

We may need to tell consumers to wrap any Message component to be announced in a div with the role, we would just need to add an example of this in our docs to point consumers to:
E.g.:
```
<div role="status">
  <Message variant="success" open {...args}>
    Form was submitted successfully. 
  </Message>
</div>
```
